spec/mailers/devise_mailer_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe NotifyMailer, type: :mailer do
+  
+  describe "user sign up" do
+    let(:user) { FactoryBot.create(:user) }
+    
+    it "should be a valid user" do
+      expect(user).to be_valid
+    end
+    
+    context "when signing up" do
+      it "send confirmation email to correct user" do   
+        response = user.send_confirmation_instructions
+        expect(response.recipients[0]).to eq user.email
+        expect(response.subject).to eq "Confirmation instructions"
+      end  
+    end
+  end
+
+  describe "reset password instructions" do
+    let(:user) { FactoryBot.create(:user) }
```

Comment:
If the user is common to all the blocks it can be floated to the top and only defined once.